tools/check_format.py

```diff
@@ -147,37 +147,61 @@ def checkNamespace(file_path):
   return []
 
 
-def fixJavaProtoOptions(file_path):
-  package_name = None
-  for line in fileinput.FileInput(file_path):
-    if line.startswith("package "):
-      result = PROTO_PACKAGE_REGEX.search(line)
-      if result is None or len(result.groups()) != 1:
-        continue
+# If the substring is not found in the file, then insert to_add
+def insertProtoOptionIfNotFound(substring, file_path, to_add):
 
-      package_name = result.group(1)
-    if "option java_package = \"io.envoyproxy.envoy" in line:
+  def repl(m):
+    return m.group(0).rstrip() + "\n\n" + to_add + "\n"
+
+  with open(file_path, "r+") as f:
+    text = f.read()
+    if substring in text:
       return []
 
-  if package_name is None:
-    return ["Unable to find package name for proto file: %s" % file_path]
+    new_text = re.sub(PROTO_PACKAGE_REGEX, repl, text)
+    f.seek(0)
+    f.write(new_text)
+    f.truncate()
+    sys.stdout.write(new_text)
+    sys.stdout.flush()
+  return []
 
-  to_add = "option java_package = \"io.envoyproxy.{}\";\n".format(package_name)
 
-  for line in fileinput.FileInput(file_path, inplace=True):
-    if line.startswith("package "):
-      line = line.replace(line, line + to_add)
-    sys.stdout.write(line)
+def packageNameForProto(file_path):
+  package_name = None
+  error_message = []
+  with open(file_path) as f:
+    result = PROTO_PACKAGE_REGEX.search(f.read())
+    if not result is None and len(result.groups()) == 1:
+      package_name = result.group(1)
+    if package_name is None:
+      error_message = ["Unable to find package name for proto file: %s" % file_path]
 
-  return []
+  return [package_name, error_message]
 
 
-def checkJavaProtoOptions(file_path):
-  for line in fileinput.FileInput(file_path):
-    if "option java_package = \"io.envoyproxy.envoy" in line:
-      return []
+def fixJavaPackageProtoOption(file_path):
+  package_name = packageNameForProto(file_path)[0]
+  to_add = "option java_package = \"io.envoyproxy.{}\";".format(package_name)
+  return insertProtoOptionIfNotFound("\noption java_package = \"io.envoyproxy.envoy", file_path,
+                                     to_add)
+
+
+# Add "option java_outer_classname = FooBarProto;" for foo_bar.proto
+def fixJavaOuterClassnameProtoOption(file_path):
+  file_name = os.path.basename(file_path)[:-len(".proto")]
+  if "-" in file_name or "." in file_name or not file_name.islower():
+    return ["Unable to decide java_outer_classname for proto file: %s" % file_path]
+
+  to_add = "option java_outer_classname = \"" \
```

Comment:
FWIW, I think if you make some of these strings like `option java_outer_classname = \"` file-level constants, it could make it easier to keep the fix/check synced.